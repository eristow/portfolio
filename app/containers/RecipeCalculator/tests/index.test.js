/**
 *
 * Tests for RecipeCalculator
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { RecipeCalculator } from '../index';

describe('<RecipeCalculator />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(<RecipeCalculator dispatch={dispatch} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<RecipeCalculator />);
    expect(firstChild).toMatchSnapshot();
  });
});
