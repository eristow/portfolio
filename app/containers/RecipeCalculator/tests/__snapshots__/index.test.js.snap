// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeCalculator /> Should render and match the snapshot 1`] = `
<div
  class="RecipeCalculator__Container-sc-1y8i9c0-0 irnkSU"
>
  <div
    class="RecipeCalculator__TotalsContainer-sc-1y8i9c0-1 dbBHhk"
  >
    <p>
      Calories: 
    </p>
    <p>
      Fat: 
    </p>
    <p>
      Carbs: 
    </p>
    <p>
      Protein: 
    </p>
    <p>
      Weight: 
    </p>
  </div>
  <div
    class="RecipeCalculator__ItemsContainer-sc-1y8i9c0-2 bfZkJA"
  >
    <div>
      <h3>
        Item 1:
      </h3>
      <input
        placeholder="Item name"
        style="width: 150px;"
        type="text"
      />
      <div>
        <p
          class="Input__Text-sc-1f9uz6z-0 rrZjn"
        >
          Calories:
        </p>
        <input
          class="Input-e90yhw-0 jVqxKD"
          min="0"
          placeholder="0"
          style="width: 150px;"
          type="number"
          value=""
        />
      </div>
      <div>
        <p
          class="Input__Text-sc-1f9uz6z-0 rrZjn"
        >
          Fat:
        </p>
        <input
          class="Input-e90yhw-0 jVqxKD"
          min="0"
          placeholder="0"
          style="width: 150px;"
          type="number"
          value=""
        />
      </div>
      <div>
        <p
          class="Input__Text-sc-1f9uz6z-0 rrZjn"
        >
          Carbs:
        </p>
        <input
          class="Input-e90yhw-0 jVqxKD"
          min="0"
          placeholder="0"
          style="width: 150px;"
          type="number"
          value=""
        />
      </div>
      <div>
        <p
          class="Input__Text-sc-1f9uz6z-0 rrZjn"
        >
          Protein:
        </p>
        <input
          class="Input-e90yhw-0 jVqxKD"
          min="0"
          placeholder="0"
          style="width: 150px;"
          type="number"
          value=""
        />
      </div>
      <div>
        <p
          class="Input__Text-sc-1f9uz6z-0 rrZjn"
        >
          Weight:
        </p>
        <input
          class="Input-e90yhw-0 jVqxKD"
          min="0"
          placeholder="0"
          style="width: 150px;"
          type="number"
          value=""
        />
      </div>
      <button
        class="Button-cbu1yy-0 hKoJdu"
        color="#03dac6"
        type="button"
      >
        Remove Item 1
      </button>
    </div>
  </div>
  <button
    class="Button-cbu1yy-0 hKoJdu"
    color="#03dac6"
    type="button"
  >
    Add Item
  </button>
</div>
`;
